### PR TITLE
Add support for the blocking HTTP client being optional (based on Cargo features).

### PR DESCRIPTION
**Pull request by vorporeal**
The blocking reqwest client doesn't exist for the `wasm32` architecture (as there's no way to implement such an API on the web).  We need to have a way to only conditionally include the blocking client so that it doesn't interfere with compiling the `warp` create to wasm.

This also fixes a compilation issue for the CLI binary that was introduced in a previous change that bumped the reqwest version to one where the blocking client was not the default.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ fn main() -> Result<(), Error> {
         .get_matches();
 
     let client = HttpClient::new(
-        reqwest::Client::new(),
+        reqwest::blocking::Client::new(),
```
Comment (szgupta) [inline src/main.rs:37]: IIUC, don't we need to conditionally choose the reqwest client based on whether or not the `blocking` feature is enabled?

**Review by szgupta — APPROVED** [review]
Just one main question.

File: src/main.rs (hunk 1 above)
Comment (vorporeal) [reply src/main.rs:37]: The `HttpClient` requires a blocking reqwest client (the APIs aren't compatible - one is standard functions and the other is async functions).

A couple notes:
1. We don't actually use the client from this crate - we use our own client that uses some of the other helper code (enums and structs).
2. This command line tool is for manually sending requests to Segment; we don't use it but I just happened to notice (when running tests) that it wasn't compiling so I fixed it here.  (I don't think anyone on our team has actually used this.)

Comment (szgupta) [reply src/main.rs:37]: Gotcha, thanks for the context

**Pull request merged**